machine-learning/02-Data Preparation and Feature Engineering.ipynb

```diff
@@ -1,45 +1,405 @@
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [
+        {
+          "data": {
+            "text/html": [
+              "<table><thead><tr><th>CanShuffle</th><th>Schema</th></tr></thead><tbody><tr><td><div class=\"dni-plaintext\">False</div></td><td><div class=\"dni-plaintext\">[ { vendor_id: String: Name: vendor_id, Index: 0, IsHidden: False, Type: { String: RawType: System.ReadOnlyMemory&lt;System.Char&gt; }, Annotations: { : Schema: [  ] } }, { rate_code: Single: Name: rate_code, Index: 1, IsHidden: False, Type: { Single: RawType: System.Single }, Annotations: { : Schema: [  ] } }, { passenger_count: Single: Name: passenger_count, Index: 2, IsHidden: False, Type: { Single: RawType: System.Single }, Annotations: { : Schema: [  ] } }, { trip_time_in_secs: Single: Name: trip_time_in_secs, Index: 3, IsHidden: False, Type: { Single: RawType: System.Single }, Annotations: { : Schema: [  ] } }, { trip_distance: Single: Name: trip_distance, Index: 4, IsHidden: False, Type: { Single: RawType: System.Single }, Annotations: { : Schema: [  ] } }, { payment_type: String: Name: payment_type, Index: 5, IsHidden: False, Type: { String: RawType: System.ReadOnlyMemory&lt;System.Char&gt; }, Annotations: { : Schema: [  ] } }, { fare_amount: Single: Name: fare_amount, Index: 6, IsHidden: False, Type: { Single: RawType: System.Single }, Annotations: { : Schema: [  ] } } ]</div></td></tr></tbody></table>"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "//Create MLContext\n",
+        "MLContext mlContext = new MLContext();\n",
+        "\n",
+        "// Create TextLoader based on the Model Input type. \n",
+        "TextLoader textLoader = mlContext.Data.CreateTextLoader<ModelInput>(separatorChar: ',', hasHeader: true);\n",
+        "\n",
+        "// Load the data into an IDataView. Load() method can support multiple files. \n",
+        "// Files must they have the same separator character, header, column names, etc. \n",
+        "IDataView data = textLoader.Load(\"data/taxi-fare.csv\");\n",
+        "\n",
+        "data.Preview(1); "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "#### Loading in memory collection\n",
+        "\n",
+        "ML.NET supports loading data from an in memory collection. This makes it easy to load from a JSON or XML file using C#. Learn how to [deserialize JSON with C#](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to?pivots=dotnet-6-0#how-to-read-json-as-net-objects-deserialize) or use [XML serializer](https://docs.microsoft.com/en-us/dotnet/api/system.xml.serialization?view=net-6.0) to get those files into memory. \n",
+        "\n",
+        "Once you have the data collection in memory, you can load it into ML.NET with the `LoadFromEnumerable` method. "
```

Comment:
Suggestion: Add link to LoadFromEnumerable method API doc

https://docs.microsoft.com/dotnet/api/microsoft.ml.dataoperationscatalog.loadfromenumerable?view=ml-dotnet